algorithms/kernel/svm/svm_train_thunder_impl.i

```diff
@@ -0,0 +1,339 @@
+
+        HelperTrainSVM<algorithmFPType, cpu>::WSSjLocal(0, nWS, KBiBlock, kdLocal, gradLocal, I, GMin, KBiBi, tau, Bj, GMax, GMax2, delta);
+
+        localDiff = GMax2 - GMin;
+
+        if (iter == 0)
+        {
+            localEps = services::internal::max<cpu, algorithmFPType>(eps, localDiff * (algorithmFPType)1e-1);
+        }
+        if (localDiff < localEps)
+        {
+            break;
+        }
+
+        const algorithmFPType yBi = yLocal[Bi];
+        const algorithmFPType yBj = yLocal[Bj];
+
+        /* Update coefficients */
+        const algorithmFPType alphaBiDelta = (yBi > 0.0f) ? C - alphaLocal[Bi] : alphaLocal[Bi];
+        const algorithmFPType alphaBjDelta = services::internal::min<cpu, algorithmFPType>((yBj > 0.0f) ? alphaLocal[Bj] : C - alphaLocal[Bj], delta);
+        delta                              = services::internal::min<cpu, algorithmFPType>(alphaBiDelta, alphaBjDelta);
+
+        /* Update alpha */
+        alphaLocal[Bi] += delta * yBi;
+        alphaLocal[Bj] -= delta * yLocal[Bj];
+
+        /* Update up/low sets */
+        char IBi = free;
+        IBi |= HelperTrainSVM<algorithmFPType, cpu>::isUpper(yBi, alphaLocal[Bi], C) ? up : free;
+        IBi |= HelperTrainSVM<algorithmFPType, cpu>::isLower(yBi, alphaLocal[Bi], C) ? low : free;
+        I[Bi] = IBi;
+
+        char IBj = free;
+        IBj |= HelperTrainSVM<algorithmFPType, cpu>::isUpper(yBj, alphaLocal[Bj], C) ? up : free;
+        IBj |= HelperTrainSVM<algorithmFPType, cpu>::isLower(yBj, alphaLocal[Bj], C) ? low : free;
+        I[Bj] = IBj;
+
+        const algorithmFPType * KBjBlock = &kernelLocal[Bj * nWS];
+        /* Update gradient */
+        PRAGMA_IVDEP
+        PRAGMA_VECTOR_ALWAYS
+        for (size_t i = 0; i < nWS; i++)
+        {
+            const algorithmFPType KiBi = KBiBlock[i];
+            const algorithmFPType KiBj = KBjBlock[i];
+            gradLocal[i] += delta * (KiBi - KiBj);
+        }
+    }
+
+    /* Compute diff and scatter to alpha vector */
+    PRAGMA_IVDEP
+    PRAGMA_VECTOR_ALWAYS
+    for (size_t i = 0; i < nWS; i++)
+    {
+        deltaAlpha[i]       = (alphaLocal[i] - oldAlphaLocal[i]) * yLocal[i];
+        alpha[wsIndices[i]] = alphaLocal[i];
+    }
+    return status;
+}
+
+template <typename algorithmFPType, typename ParameterType, CpuType cpu>
+services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::updateGrad(const algorithmFPType * kernelWS,
+                                                                                        const algorithmFPType * deltaalpha, algorithmFPType * grad,
+                                                                                        const size_t nVectors, const size_t nWS)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(updateGrad);
+    char trans  = 'N';
+    DAAL_INT m_ = nVectors;
+    DAAL_INT n_ = nWS;
+    algorithmFPType alpha(1.0);
+    DAAL_INT lda = m_;
+    DAAL_INT incx(1);
+    algorithmFPType beta(1.0);
+    DAAL_INT incy(1);
```

Comment:
incY. At least to avoid confusion ('incy' is valid word).

Reply:
Applied